sfdc/modules/bulk/tests/bulk_csv_delete.bal

```diff
@@ -19,125 +19,115 @@ import ballerina/test;
 import ballerina/lang.runtime;
 
 @test:AfterSuite {}
-function deleteCsv() {
+function deleteCsv() returns error? {
     log:printInfo("baseClient -> deleteCsv");
     string batchId = "";
-    string contacts = getCsvContactsToDelete(csvQueryResult);
 
     //create job
-    error|BulkJob deleteJob = baseClient->createJob("delete", "Contact", "CSV");
+    BulkJob deleteJob = check baseClient->createJob("delete", "Contact", "CSV");
 
-    if (deleteJob is BulkJob) {
-        //add csv content
-        foreach var i in 1 ..< maxIterations {
-            error|BatchInfo batch = baseClient->addBatch(deleteJob, <@untainted>contacts);
-            if (batch is BatchInfo) {
-                test:assertTrue(batch.id.length() > 0, msg = "Could not upload the contacts to delete using CSV.");
-                batchId = batch.id;
-                break;
+    //add csv content
+    foreach var i in 1 ..< maxIterations {
+        error|BatchInfo batch = baseClient->addBatch(deleteJob, csvInputResult);
+        if (batch is BatchInfo) {
+            test:assertTrue(batch.id.length() > 0, msg = "Could not upload the contacts to delete using CSV.");
+            batchId = batch.id;
+            break;
+        } else {
+            if i != 5 {
+                log:printWarn("addBatch Operation Failed! Retrying...");
+                runtime:sleep(delayInSecs);
```

Comment:
Where do we define `delayInSecs`? I mean into the test cases how do we inject that value?

Reply:
The `delayInSecs` is added in  https://github.com/ballerina-platform/module-ballerinax-sfdc/blob/f72db37e33688bc800ddfc4fd065c85ab6a2157d/sfdc/modules/bulk/tests/bulk_csv_insert.bal#L23 and we reused this in all the places.

Reply:
OK. That is fine.